dia4_Challenge.sql

```diff
@@ -0,0 +1,136 @@
+-- FUNCIONES
+
+-- 1) Write a query to fetch even numbered records from employees table. 
+SELECT EMPLOYEE_ID, FIRST_NAME, LAST_NAME FROM Employees
+	where EMPLOYEE_ID % 2 = 0;
+
+
+-- 2) Write a query to find the 5th maximum salary in the employees table. 
+SELECT DISTINCT SALARY 
+	FROM Employees e1 
+	WHERE 4 = (SELECT COUNT(DISTINCT SALARY) 
+	FROM Employees e2 
+	WHERE e2.salary >= e1.salary);
+
+-- 3) Write a query to find the 4th minimum salary in the employees table. 
+SELECT DISTINCT SALARY 
+	FROM Employees e1 
+	WHERE 2 = (SELECT COUNT(DISTINCT SALARY) 
+	FROM Employees e2 
+	WHERE e2.salary <= e1.salary);
+
+-- 4) Write a query to select last 10 records from the employees table. 
+SELECT EMPLOYEE_ID, FIRST_NAME, LAST_NAME FROM Employees
+	ORDER BY EMPLOYEE_ID DESC
+	LIMIT 10;
+
+-- 5) Write a query to get the 3 maximum salaries. 
+SELECT EMPLOYEE_ID, FIRST_NAME, LAST_NAME, SALARY FROM Employees
+	ORDER BY SALARY
+	DESC LIMIT 3;
+
+-- 6) Write a query to get the 3 minimum salaries. 
+SELECT EMPLOYEE_ID, FIRST_NAME, LAST_NAME, SALARY FROM Employees
+	ORDER BY SALARY
+	LIMIT 3;
+
+
+-- SUB-QUERIES
+-- 1) Write a query to find the name (FIRST_NAME, LAST_NAME) and salary of the employees who earn a salary that is higher than the salary of all the Shipping Clerks (JOB_ID = 'SH_CLERK'). Sort the results of the salary of the lowest to highest.
+
+SELECT FIRST_NAME, LAST_NAME, SALARY FROM Employees
+	WHERE SALARY > ( 
+        SELECT MAX(SALARY) FROM Employees 
+        WHERE JOB_ID = 'SH_CLERK')
+        ORDER BY SALARY;
+
+-- 2) Write a query to find the name (FIRST_NAME, LAST_NAME) of the employees who are not managers.
+
+SELECT FIRST_NAME, LAST_NAME, JOB_ID FROM Employees
```

Comment:
Está casi correcto. Este query regresa a los presidentes, que también son managers. Recuerda que tienes acceso a la lista de managers en la misma tabla de empleados.

Reply:
Corregido:
SELECT FIRST_NAME, LAST_NAME, JOB_ID FROM Employees
	WHERE JOB_ID IN (
	SELECT DISTINCT JOB_ID from Employees
		WHERE JOB_ID NOT LIKE '%MAN'
		AND JOB_ID NOT LIKE '%PRES'
	);